R/PipeOpLearnerCV.R

```diff
@@ -159,53 +165,71 @@ PipeOpLearnerCV = R6Class("PipeOpLearnerCV",
       } else {
         multiplicity_recurse(self$state, clone_with_state, learner = private$.learner)
       }
+    },
+    resampling = function(val) {
+      if (!missing(val)) {
+        if (!identical(val, private$.resampling)) {
+          stop("$resampling is read-only.")
+        }
+      }
+      private$.resampling
     }
   ),
   private = list(
     .train_task = function(task) {
       on.exit({private$.learner$state = NULL})
-
-      # Train a learner for predicting
+      # train a learner for predicting
       self$state = private$.learner$train(task)$state
-      pv = private$.crossval_param_set$values
 
-      # Compute CV Predictions
-      if (pv$method != "insample") {
-        rdesc = mlr_resamplings$get(pv$method)
-        if (pv$method == "cv") rdesc$param_set$values = list(folds = pv$folds)
-        rr = resample(task, private$.learner, rdesc)
-        prds = as.data.table(rr$prediction(predict_sets = "test"))
-      } else {
-        prds = as.data.table(private$.learner$predict(task))
-      }
+      # compute resampled predictions
+      rr = resample(task, private$.learner, private$.resampling)
```

Comment:
is there a way to check if the resampling fits a model on all the data, which could then be used for prediction without needing to fit twice?